Add compatibility with py311

ssbarnea opened the pull request:
This is still marked as draft because I did not test it enough. Somehow I have the impression that the current test coverage is not extensive, so I need bit of manual testing before being confident to make it ready.

ssbarnea commented:
@cansarigol You need to make some changes to allow automatic testing in a safe way.

Move the `POETRY_PYPI_TOKEN_PYPI` from repository secrets into an new environment calles `release` and add the `environment: release` to the release pipeline. You can see an example at https://github.com/ansible/ansible-lint/blob/main/.github/workflows/release.yml#L11

That will prevent the secret from being used in testing pipelines and you will not need to manually approve GHA execution.

cansarigol commented:
@ssbarnea I added py3.11 in the python version of test action and fixed a bit runner. you can rebase it. I want to keep release and test in two different actions. for release, action is focused on tags. because of that, I put it like this.

ssbarnea commented:
> @ssbarnea I added py3.11 in the python version of test action and fixed a bit runner. you can rebase it. I want to keep release and test in two different actions. for release, action is focused on tags. because of that, I put it like this.

I also have the release in a different workflow file, for the same reason, better to keep test decoupled from release, security and maintenance reasons.
